### PR TITLE
Port System.Reflection.Metadata source code comments to Docs (Method*-T files)

### PR DESCRIPTION
**Pull request by carlossanlop**
## Summary

Automatically ported triple slash source code comments found in System.Reflection.Metadata that were not yet documented in dotnet-api-docs.

This PR contains changes to filenames starting with Method*-T.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Comment by carlossanlop** [comment]
@mairaw @rpetrusha please take a look whenever possible.

Adding the area owners @tmat, @nguerrera in case they have any suggestions or remarks about these changes.

**Comment by carlossanlop** [comment]
@mairaw @rpetrusha if this looks good, can we get it merged?

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 1]
```diff
@@ -54,10 +54,13 @@
         <Parameter Name="peReader" Type="System.Reflection.PortableExecutable.PEReader" RefType="this" />
       </Parameters>
       <Docs>
-        <param name="peReader">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="peReader"><see langword="this" /> <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <param name="peReader">The current <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 2]
```diff
@@ -54,10 +54,13 @@
...
+        <summary>Gets a <see cref="T:System.Reflection.Metadata.MetadataReader" /> from a <see cref="T:System.Reflection.PortableExecutable.PEReader" />.</summary>
+        <returns>A <see cref="T:System.Reflection.Metadata.MetadataReader" /> instance.</returns>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <returns>A metdata reader.</returns>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 3]
```diff
@@ -84,11 +87,14 @@
         <Parameter Name="options" Type="System.Reflection.Metadata.MetadataReaderOptions" />
       </Parameters>
       <Docs>
-        <param name="peReader">To be added.</param>
-        <param name="options">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="peReader"><see langword="this" /> <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
+        <param name="options">A <see cref="F:System.Reflection.Metadata.MetadataReaderOptions" /> enum value indicating the metadata reading configuration.</param>
+        <summary>Gets a <see cref="T:System.Reflection.Metadata.MetadataReader" /> from a <see cref="T:System.Reflection.PortableExecutable.PEReader" />.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <summary>Gets a  metadata reader with the specified metadata reading configuration from a <see cref="T:System.Reflection.PortableExecutable.PEReader" />.</summary>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 4]
```diff
@@ -84,11 +87,14 @@
...
+        <returns>A <see cref="T:System.Reflection.Metadata.MetadataReader" /> instance.</returns>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <returns>A  metadata reader with the specified metadata reading configuration.</returns>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 5]
```diff
@@ -84,11 +87,14 @@
         <Parameter Name="options" Type="System.Reflection.Metadata.MetadataReaderOptions" />
       </Parameters>
       <Docs>
-        <param name="peReader">To be added.</param>
-        <param name="options">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="peReader"><see langword="this" /> <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <param name="peReader">The current <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 6]
```diff
@@ -84,11 +87,14 @@
...
+        <param name="options">A <see cref="F:System.Reflection.Metadata.MetadataReaderOptions" /> enum value indicating the metadata reading configuration.</param>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <param name="options">An  enumeration value indicating the metadata reading configuration.</param>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 7]
```diff
@@ -116,12 +122,16 @@
...
       <Docs>
-        <param name="peReader">To be added.</param>
-        <param name="options">To be added.</param>
-        <param name="utf8Decoder">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="peReader"><see langword="this" /> <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
+        <param name="options">A <see cref="F:System.Reflection.Metadata.MetadataReaderOptions" /> enum value indicating the metadata reading configuration.</param>
+        <param name="utf8Decoder">A <see cref="T:System.Reflection.Metadata.MetadataStringDecoder" /> instance with the encoding configuration.</param>
+        <summary>Gets a <see cref="T:System.Reflection.Metadata.MetadataReader" /> from a <see cref="T:System.Reflection.PortableExecutable.PEReader" />.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <summary>Gets a metadata reader with the specified metadata reading configuration and encoding configuration from a <see cref="T:System.Reflection.PortableExecutable.PEReader" />.</summary>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 8]
```diff
@@ -116,12 +122,16 @@
         <Parameter Name="utf8Decoder" Type="System.Reflection.Metadata.MetadataStringDecoder" />
       </Parameters>
       <Docs>
-        <param name="peReader">To be added.</param>
-        <param name="options">To be added.</param>
-        <param name="utf8Decoder">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="peReader"><see langword="this" /> <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <param name="peReader">The current <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 9]
```diff
@@ -116,12 +122,16 @@
...
+        <param name="options">A <see cref="F:System.Reflection.Metadata.MetadataReaderOptions" /> enum value indicating the metadata reading configuration.</param>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <param name="options">An enumeration value indicating the metadata reading configuration.</param>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 10]
```diff
@@ -116,12 +122,16 @@
...
+        <param name="utf8Decoder">A <see cref="T:System.Reflection.Metadata.MetadataStringDecoder" /> instance with the encoding configuration.</param>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <param name="utf8Decoder">A metadata string decoder with the encoding configuration.</param>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 11]
```diff
@@ -116,12 +122,16 @@
...
+        <summary>Gets a <see cref="T:System.Reflection.Metadata.MetadataReader" /> from a <see cref="T:System.Reflection.PortableExecutable.PEReader" />.</summary>
+        <returns>A <see cref="T:System.Reflection.Metadata.MetadataReader" /> instance.</returns>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <returns>>A metadata reader with the specified metadata reading configuration and encoding configuration.</returns>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 12]
```diff
@@ -148,11 +158,15 @@
         <Parameter Name="relativeVirtualAddress" Type="System.Int32" />
       </Parameters>
       <Docs>
-        <param name="peReader">To be added.</param>
-        <param name="relativeVirtualAddress">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="peReader"><see langword="this" /> <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <param name="peReader">The current <see cref="T:System.Reflection.PortableExecutable.PEReader" /> instance.</param>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 13]
```diff
@@ -148,11 +158,15 @@
...
+        <param name="relativeVirtualAddress">A <see cref="T:System.Int32" /> value indicating the Relative Virtual Address (RVA).</param>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <param name="relativeVirtualAddress">The Relative Virtual Address (RVA).</param>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 14]
```diff
@@ -148,11 +158,15 @@
...
+        <summary>Returns a body block of a method with the specified Relative Virtual Address (RVA);</summary>
+        <returns>A <see cref="T:System.Reflection.Metadata.MethodBodyBlock" /> instance.</returns>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <returns>A method block body instance.</returns>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 15]
```diff
@@ -148,11 +158,15 @@
...
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException"><paramref name="peReader" /> is <see langword="null" />.</exception>
+        <exception cref="T:System.BadImageFormatException">The body is not found in the metadata or is invalid.</exception>
+        <exception cref="T:System.InvalidOperationException">Section where the method is stored is not available.</exception>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <exception cref="T:System.InvalidOperationException">The section where the method is stored is not available.</exception>
```

File: xml/System.Reflection.Metadata/PEReaderExtensions.xml [hunk 16]
```diff
@@ -148,11 +158,15 @@
...
+        <exception cref="T:System.IO.IOException">IO error while reading from the underlying stream.</exception>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/PEReaderExtensions.xml]:
```suggestion
        <exception cref="T:System.IO.IOException">IO error occurred while reading from the underlying stream.</exception>
```

File: xml/System.Reflection.Metadata/ReservedBlob`1.xml [hunk 17]
```diff
@@ -104,8 +104,8 @@
         <ReturnType>THandle</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>The reserved blob handle of type <typeparamref name="THandle" />.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/ReservedBlob`1.xml]:
```suggestion
        <summary>Gets the reserved blob handle.</summary>
```

File: xml/System.Reflection.Metadata/ReservedBlob`1.xml [hunk 18]
```diff
@@ -104,8 +104,8 @@
...
+        <value>A <typeparamref name="THandle" /> instance.</value>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/ReservedBlob`1.xml]:
```suggestion
        <value>The reserved bloc handle.</value>
```

File: xml/System.Reflection.Metadata/ReservedBlob`1.xml [hunk 19]
```diff
@@ -80,8 +80,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Returns a <see cref="T:System.Reflection.Metadata.BlobWriter" /> to be used to update the content.</summary>
+        <returns>A <see cref="T:System.Reflection.Metadata.BlobWriter" /> instance.</returns>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/ReservedBlob`1.xml]:
```suggestion
        <returns>A blob writer to be used to update the content.</returns>
```

File: xml/System.Reflection.Metadata/StandaloneSignature.xml [hunk 20]
```diff
@@ -230,8 +231,8 @@
         <ReturnType>System.Reflection.Metadata.BlobHandle</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Gets a handle to the signature blob.</summary>
+        <value>A <see cref="T:System.Reflection.Metadata.BlobHandle" /> instance.</value>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/StandaloneSignature.xml]:
```suggestion
        <value>A handle to the signature blob.</value>
```

File: xml/System.Reflection.Metadata/StandaloneSignature.xml [hunk 21]
```diff
@@ -204,9 +204,10 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Determines the kind of signature, which can be <see cref="F:System.Reflection.Metadata.SignatureKind.Method" /> or <see cref="F:System.Reflection.Metadata.SignatureKind.LocalVariables" />.</summary>
+        <returns>A <see cref="F:System.Reflection.Metadata.StandaloneSignatureKind" /> enum value.</returns>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/StandaloneSignature.xml]:
```suggestion
        <returns>An enumeration value that indicates the signature kind.</returns>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 22]
```diff
@@ -75,8 +75,8 @@
         <ReturnType>System.Reflection.Metadata.EntityHandle</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>The base type of the type definition: either <see cref="T:System.Reflection.Metadata.TypeSpecificationHandle" />, <see cref="T:System.Reflection.Metadata.TypeReferenceHandle" /> or <see cref="T:System.Reflection.Metadata.TypeDefinitionHandle" />.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <summary>Gets the base type of the type definition: either <see cref="T:System.Reflection.Metadata.TypeSpecificationHandle" />, <see cref="T:System.Reflection.Metadata.TypeReferenceHandle" /> or <see cref="T:System.Reflection.Metadata.TypeDefinitionHandle" />.</summary>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 23]
```diff
@@ -75,8 +75,8 @@
...
+        <value>An <see cref="T:System.Reflection.Metadata.EntityHandle" /> instance.</value>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <value>The base type of the type definition.</value>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 24]
```diff
@@ -156,8 +156,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Returns the enclosing type of a specified nested type. Otherwise, if the type is not nested, it returns a a handle where the <see cref="P:System.Reflection.Metadata.TypeDefinitionHandle.IsNil" /> property is <see langword="true" />.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <summary>Returns the enclosing type of a specified nested type. </summary>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 25]
```diff
@@ -156,8 +156,8 @@
...
+        <returns>A <see cref="T:System.Reflection.Metadata.TypeDefinitionHandle" /> instance.</returns>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <returns>The enclosing type of the specified nested type, or a handle a handle whose <see cref="P:System.Reflection.Metadata.TypeDefinitionHandle.IsNil" /> property is <see langword="true" /> if the type is not nested.</returns>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 26]
```diff
@@ -372,8 +372,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Returns an array of types nested in the specified type.</summary>
+        <returns>An immutable array of type definition handles.</returns>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <returns>An immutable array of type definition handles that represent types nested in the specified type.</returns>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 27]
```diff
@@ -421,8 +421,8 @@
         <ReturnType>System.Boolean</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Indicates whether this is a nested type.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <summary>Gets a value that indicates whether this is a nested type.</summary>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 28]
```diff
@@ -447,8 +447,8 @@
         <ReturnType>System.Reflection.Metadata.StringHandle</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Name of the type.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <summary>Gets the name of the type.</summary>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 29]
```diff
@@ -447,8 +447,8 @@
...
+        <value>A <see cref="T:System.Reflection.Metadata.StringHandle" /> instance.</value>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <value>The name of the type.</value>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 30]
```diff
@@ -473,8 +473,8 @@
         <ReturnType>System.Reflection.Metadata.StringHandle</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Full name of the namespace where the type is defined. Otherwise, if the type is nested or defined in a root namespace, it returns a handle where the <see cref="P:System.Reflection.Metadata.StringHandle.IsNil" /> property is <see langword="true" />.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <summary>Gets the full name of the namespace where the type is defined. </summary>
```

File: xml/System.Reflection.Metadata/TypeDefinition.xml [hunk 31]
```diff
@@ -473,8 +473,8 @@
...
+        <value>A <see cref="T:System.Reflection.Metadata.StringHandle" /> instance.</value>
```
Comment (rpetrusha) [inline xml/System.Reflection.Metadata/TypeDefinition.xml]:
```suggestion
        <value>The full name of the namespace where the type is defined, or a handle whose <see cref="P:System.Reflection.Metadata.StringHandle.IsNil" /> property is <see langword="true" /> if the type is nested or defined in a root namespace.</value>
```

**Comment by carlossanlop** [comment]
@rpetrusha Thank you for your suggestions. I applied them all. If the build passes, can we get this merged?

**Comment by rpetrusha** [comment]
Thanks for the additional changes, @carlossanlop. I'll merge your PR now.

**Pull request merged**